ui/ui-components/pages/app/[id]/refresh.tsx

```diff
@@ -15,10 +15,62 @@ import AppRefreshQueryScheduleList from "../../../components/app/AppRefreshSched
 import AppRefreshQueryStats from "../../../components/app/AppRefreshQueryStats";
 import { Actions, Models } from "../../../utils/apiData";
 import { grouparooUiEdition } from "../../../utils/uiEdition";
-import { NextPageContext } from "next";
 import { generateClient } from "../../../client/client";
+import { withServerErrorHandler } from "../../../utils/withServerErrorHandler";
+import { NextPageWithInferredProps } from "../../../utils/pageHelper";
 
-export default function Page(props) {
+export const getServerSideProps = withServerErrorHandler(async (ctx) => {
+  const { id } = ctx.query;
+  const client = generateClient(ctx);
+  const { app } = await client.request<Actions.AppView>("get", `/app/${id}`);
+
+  let foundAppRefreshQuery: Models.AppRefreshQueryType;
+
+  if (app.appRefreshQuery !== null) {
+    const { appRefreshQuery } =
+      await client.request<Actions.AppRefreshQueryView>(
+        "get",
+        `/appRefreshQuery/${app.appRefreshQuery.id}`
+      );
+    foundAppRefreshQuery = appRefreshQuery;
+  }
+
+  const { sources } = await client.request<Actions.SourcesList>(
+    "get",
+    `/sources`
+  );
+
+  let scheduleRuns = [];
+  let schedules = [];
+
+  if (sources) {
+    schedules = sources
+      .filter(
+        (source) => source.appId === app.id && source.schedule?.refreshEnabled
+      )
+      .map((source) => source.schedule);
```

Comment:
Nitpick, and I'm not sure if this is more efficient.
```suggestion
  const scheduleRuns = [];
  const schedules = [];

  if (sources) {
    schedules.push(...sources
      .filter(
        (source) => source.appId === app.id && source.schedule?.refreshEnabled
      )
      .map((source) => source.schedule));
```